code/app/src/main/java/com/example/sigma_blue/entity/item/Item.java

```diff
@@ -342,13 +344,13 @@ public List<Tag> getTags() {
     }
 
     /**
-     * Returns the list of tag names
+     * Returns the list of document tag ids.
      *
-     * @return
+     * @return a list of the document tag ID
      */
-    public ArrayList<String> getTagNames() {
-        return new ArrayList<>(tags.stream().map(Tag::getDocID).collect(Collectors
-                .toList()));
+    public ArrayList<String> getTagDocIDs() {
```

Comment:
this is a bad name, it stores an array of strings. each string is used to rebuild a tag object from scratch (there is no second query to lookup each tag).